rpcclient/mocks_test.go

```diff
@@ -1,7 +1,7 @@
 // Copyright 2023 The AthanorLabs/atomic-swap Authors
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package rpc
+package rpcclient
```

Comment:
I moved the tests, mostly unchanged, from `rpc` to `rpcclient` to avoid a circular dependency.